WinUIB2CApp/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,248 @@
+
+                // Must be called from UI thread
+                this.AuthResult = await this.PublicClientApplication.AcquireTokenInteractive(scopes)
+                    .WithLoginHint(existingUser?.Username ?? String.Empty)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+            catch (MsalException msalEx)
+            {
+                Debug.WriteLine($"Error Acquiring Token:{Environment.NewLine}{msalEx}");
+            }
+
+            return this.AuthResult.AccessToken;
+        }
+
+        /// <summary>
+        /// Signs the in user and acquire access token for a provided set of scopes.
+        /// </summary>
+        /// <param name="scopes">The scopes.</param>
+    /// <param name="extraclaims">The extra claims, usually from CAE. We basically handle CAE by sending the user back to Azure AD for additional processing and requesting a new access token</param>
+        /// <returns></returns>
+        public async Task<String> SignInUserAndAcquireAccessToken(string[] scopes, string extraclaims)
+        {
+            var existingUser = await FetchSignedInUserFromCache().ConfigureAwait(false);
+
+            try
+            {
+                // Send the user to Azure AD for re-authentication as a silent acquisition wont resolve any CAE scenarios
+                this.AuthResult = await PublicClientApplication.AcquireTokenInteractive(scopes)
+                    .WithLoginHint(existingUser?.Username ?? String.Empty)
+                    .WithClaims(extraclaims)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+            catch (MsalException msalEx)
+            {
+                Debug.WriteLine($"Error Acquiring Token interactively:{Environment.NewLine}{msalEx}");
+            }
+
+            return this.AuthResult.AccessToken;
+        }
+
+        /// <summary>
+        /// Shows a pattern to sign-in a user interactively in applications that are input constrained and would need to fall-back on device code flow.
+        /// </summary>
+        /// <param name="scopes">The scopes.</param>
+        /// <param name="existingAccount">The existing account.</param>
+        /// <returns></returns>
+        private async Task<AuthenticationResult> SignInUserInteractivelyAsync(string[] scopes, IAccount existingAccount = null)
+        {
+            // If the operating system has UI
+            if (this.PublicClientApplication.IsUserInteractive())
+            {
+                return await this.PublicClientApplication.AcquireTokenInteractive(scopes)
+                    .WithLoginHint(existingAccount?.Username ?? String.Empty)
+                    .WithParentActivityOrWindow(windowHandle) // TODO: fix it
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+
+            // If the operating system does not have UI (e.g. SSH into Linux), you can fallback to device code, however this
```

Comment:
This is a WinUI app, so it only runs on Windows. 
But it's a good point that we should have a cross-plat app (e.g. console app) showcasing this as well. Definitely for B2C it would be better, as there is no broker.